lib/datasets/zip_writer.rb

```diff
@@ -10,45 +11,40 @@ class ZipWriter
   # @yield [input_zip, output_zip] optional block for implementation of copying
   # from input zip to output zip. By default gets all txt files in input zip
   # and copies them to output zip.
-  def write(src_path, dest_path)
-    # open the source and destination zips
-
-    temp_zip = Tempfile.open(['dataset', '.zip'], File.dirname(dest_path))
-
-    begin
-      # initialize zip file - required since we opened the file w/ Tempfile
-      Zip::OutputStream.open(temp_zip) { |zos| }
-
+  def write(src_path, dest_path, &block)
+    Tempfile.open(['dataset', '.zip'], dest_path.dirname) do |temp_zip|
       Zip::File.open(src_path) do |input_zip|
         Zip::File.open(temp_zip.path, Zip::File::CREATE) do |output_zip|
-          if block_given?
-            yield input_zip, output_zip
-          else
-            copy_text(input_zip, output_zip)
-          end
+          (block || copy_text).call input_zip, output_zip
```

Comment:
Logically the same as before, but seemed cleaner to me.  The `#copy_text` definition was changed to define a block that provides its previous functionality.